src/v/pandaproxy/schema_registry/store.h

```diff
@@ -149,6 +135,55 @@ class store {
         return res;
     }
 
+    ///\brief Return the value of the 'deleted' field on a subject
+    result<is_deleted> is_subject_deleted(const subject& sub) const {
+        auto sub_it = BOOST_OUTCOME_TRYX(
+          get_subject_iter(sub, include_deleted::yes));
+        return sub_it->second.deleted;
+    }
+
+    ///\brief Return the value of the 'deleted' field on a subject
+    result<std::vector<seq_marker>>
+    get_subject_written_at(const subject& sub) const {
+        auto sub_it = BOOST_OUTCOME_TRYX(
+          get_subject_iter(sub, include_deleted::yes));
+
+        if (!sub_it->second.deleted) {
+            // Refuse to yield sequence history for anything that
+            // hasn't been soft-deleted, to prevent a hard-delete
+            // from generating tombstones without a preceding soft-delete
+            return not_deleted(sub);
+        } else {
+            return sub_it->second.written_at;
+        }
+    }
+
+    ///\brief Return the value of the 'deleted' field on a subject
+    result<std::vector<seq_marker>> get_subject_version_written_at(
```

Comment:
Is `vector` correct here? Or is an `optional<schema_version>` more appropriate? Can this fail to produce a version? I think there's an assert to the contrary at the call site. Should emit a different failure type such as `subject_version_not_found` in that case?

Reply:
You would expect two seq_markers here -- one for creating the version and one for soft deleting it (this function is only used when generating tombstones during a permanent delete)

There can also have just one seq_marker if the soft delete was done on the whole subject, then later the hard delete is done on a particular version.

The vassert was correct in that an empty list here is a bug, but it's probably overkill to crash redpanda for it  given that we could just fail the request, so I've changed it to raise not_found errors in the case of no seq_markers.